contrib/NeonVM/bytecode.c

```diff
@@ -15,59 +15,140 @@
 #include "nstring.h"
 #include "util.h"
 
-static uint16_t get_uint16(const uint8_t *pobj, size_t nBuffSize, uint32_t *i)
-{
-    if (*i+2 > nBuffSize) {
-        fatal_error("Bytecode exception: Read past EOF.");
-    }
-    uint16_t r = (pobj[*i] << 8) | pobj[*i+1];
-    *i += 2;
-    return r;
-}
+//static uint16_t get_uint16(const uint8_t *pobj, size_t nBuffSize, uint32_t *i)
+//{
+//    if (*i+2 > nBuffSize) {
+//        fatal_error("Bytecode exception: Read past EOF.");
+//    }
+//    uint16_t r = (pobj[*i] << 8) | pobj[*i+1];
+//    *i += 2;
+//    return r;
+//}
+//
+//static uint32_t get_uint32(const uint8_t *pobj, size_t nBuffSize, uint32_t *i)
+//{
+//    if (*i+4 > nBuffSize) {
+//        fatal_error("Bytecode excpetion: Read past EOF.");
+//    }
+//    uint32_t r = (pobj[*i] << 24) | (pobj[*i+1] << 16) | (pobj[*i+2] << 8) | pobj[*i+3];
+//    *i += 4;
+//    return r;
+//}
 
-static uint32_t get_uint32(const uint8_t *pobj, size_t nBuffSize, uint32_t *i)
+unsigned int get_vint(const uint8_t *pobj, size_t nBuffSize, size_t *i)
 {
-    if (*i+4 > nBuffSize) {
-        fatal_error("Bytecode excpetion: Read past EOF.");
+    unsigned int r = 0;
+    while (*i < nBuffSize) {
+        unsigned int x = pobj[*i];
```

Comment:
It would perhaps be clearer here to use `unsigned char`. Your code is not wrong as written, but I think that would improve the readability.